Podcast/InternalProfileHeaderView.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  InternalProfileHeaderView.swift
+//  Podcast
+//
+//  Created by Drew Dunne on 3/7/17.
+//  Copyright © 2017 Cornell App Development. All rights reserved.
+//
+
+import UIKit
+
+protocol InternalProfileHeaderViewDelegate {
+    func internalProfileHeaderViewDidPressViewProfile(internalProfileHeaderView: InternalProfileHeaderView)
+}
+
+class InternalProfileHeaderView: UIView {
+    
+    static let height: CGFloat = 96
+    
+    let height: CGFloat = InternalProfileHeaderView.height
+    let profileImageSideLength: CGFloat = 60
+    let profileImageX: CGFloat = 18
+    let padding: CGFloat = 12
+    let nameLabelY: CGFloat = 31
+    let nameLabelHeight: CGFloat = 17
+    let viewProfileLabelY: CGFloat = 51
+    let viewProfileLabelHeight: CGFloat = 14.5
+    let separatorHeight: CGFloat = 1
+    
+    var viewProfileButton: UIButton!
+    var profileImage: UIImageView!
+    var nameLabel: UILabel!
+    var viewProfileLabel: UILabel!
+    var separator: UIView!
+    
+    var delegate: InternalProfileHeaderViewDelegate?
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        backgroundColor = .podcastWhite
+        
+        profileImage = UIImageView(frame: .zero)
+        profileImage.layer.cornerRadius = profileImageSideLength / 2
+        profileImage.clipsToBounds = true
+        profileImage.backgroundColor = .podcastWhiteDark
+        addSubview(profileImage)
+        
+        nameLabel = UILabel(frame: .zero)
+        nameLabel.font = .systemFont(ofSize: 14, weight: UIFontWeightSemibold)
+        nameLabel.textColor = .podcastBlack
+        nameLabel.textAlignment = .left
+        addSubview(nameLabel)
+        
+        viewProfileLabel = UILabel(frame: .zero)
+        viewProfileLabel.font = .systemFont(ofSize: 12, weight: UIFontWeightRegular)
+        viewProfileLabel.textColor = .podcastTealBackground
+        viewProfileLabel.textAlignment = .left
+        viewProfileLabel.text = "View your profile"
+        addSubview(viewProfileLabel)
+        
+        separator = UIView(frame: .zero)
+        separator.backgroundColor = .podcastGray
+        addSubview(separator)
+        
+        viewProfileButton = UIButton(type: .custom)
```

Comment:
why does this button not have a frame? also why is there a viewprofile label and button? why not just a button?